### DISCUSSION TITLE
Error: Option '-h' requires 2 arguments.

### DISCUSSION THREAD

**Discussion by hack123mayank** [projectdiscovery/httpx#1299] [General]
When you use this command after installing the app- **httpx -h** 
Sometimes this type of error occurs **{-Error: Option '-h' requires 2 arguments.}** 
If you are encountering this error at the moment, please update your Kali system first, then perform an upgrade. Afterward, restart your Kali system, and this error should no longer occur.

**Comment by MetzinAround** [comment]
Thanks for the info!

**Comment by aleeeeks** [comment]
I had a similar issue.
It turned out that the package manager (apt) installed the python3-httpx on the system. So when you run httpx -h, the command that is being executed is not related to the ProjectDiscovery's httpx.
You should uninstall and purge data, which is related to apt python3-httpx, and clear cache.
Reinstall projectdiscovery httpx, and check if you have the latest version of Go (if you install it with Go).

**Reply by glaucusec** [reply]
you are right. 
`which httpx` returned /usr/bin/httpx. I deleted the it using `sudo rm /usr/bin/httpx`.
now `which httpx` will return the correct httpx, something like this -` /home/username/go/bin/httpx`

**Comment by olearycrew** [comment]
See https://github.com/projectdiscovery/httpx/discussions/1587
